akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala

```diff
@@ -171,38 +171,49 @@ object CoordinatedShutdown extends ExtensionId[CoordinatedShutdown] with Extensi
     coord
   }
 
+  // locate reason-specifi overrides and merge with defaults.
+  private[akka] def confWithOverrides(conf: Config, reason: Option[Reason]): Config = {
+    reason.flatMap { r ⇒
+      val basePath = s"""reason-overrides."${r.getClass.getCanonicalName}""""
```

Comment:
I'm not sure we should use `getCanonicalName`. It returns `null` for some cases and:

```
scala> object Foo { class Bar}

scala> classOf[Foo.Bar].getName
res4: String = Foo$Bar

scala> classOf[Foo.Bar].getCanonicalName
java.lang.InternalError: Malformed class name
  at java.lang.Class.getSimpleName(Class.java:1330)
  at java.lang.Class.getCanonicalName(Class.java:1399)
  at java.lang.Class.getCanonicalName(Class.java:1396)
```